File: Examples/test-suite/matlab/using_extend_runme.m
import using_extend.*

f = FooBar();
if (f.blah(3) ~= 3)
    error('blah(int)')
end

if (f.blah(3.5) ~= 3.5)
    error('blah(double)')
end

if (~strcmp(f.blah('hello'),'hello'))
    error('blah(char *)')
end

if (f.blah(3,4) ~= 7)
    error('blah(int,int)')
end

if (f.blah(3.5,7.5) ~= (3.5+7.5))
    error('blah(double,double)')
end


if (f.duh(3) ~= 3)
    error('duh(int)')
end

File: Examples/test-suite/javascript/using_extend_runme.js
var using_extend = require("using_extend");

f = new using_extend.FooBar();
if (f.blah(3) != 3) {
    throw new Error("blah(int)");
}

if (f.blah(3.5) != 3.5) {
    throw new Error("blah(double)");
}

if (f.blah("hello") != "hello") {
    throw new Error("blah(char *)");
}

if (f.blah(3, 4) != 7) {
    throw new Error("blah(int,int)");
}

if (f.blah(3.5, 7.5) != (3.5 + 7.5)) {
    throw new Error("blah(double,double)");
}


if (f.duh(3) != 3) {
    throw new Error("duh(int)");
}
